quickwit/quickwit-serve/src/lib.rs

```diff
@@ -760,7 +760,9 @@ pub async fn serve_quickwit(
         // We must decommission the ingester first before terminating the indexing pipelines that
         // may consume from it. We also need to keep the gRPC server running while doing so.
         if let Some(ingester) = ingester_opt {
-            if let Err(error) = wait_for_ingester_decommission(ingester).await {
+            if let Err(error) =
+                wait_for_ingester_decommission(ingester, Duration::from_secs(5)).await
```

Comment:
5s is arbitrary and probably a bit short. We could also only timeout when we fail to reach the CP/metastore and keep retrying for other kinds of errors